### ISSUE TITLE
[bug] `go-aws-sso refresh` raises `panic: runtime error: invalid memory address or nil pointer dereference`

### ISSUE THREAD

**Issue by PaarthShah** [theurichde/go-aws-sso#178]
**Describe the bug**
Running `go-aws-sso refresh` does not work, even if `go-aws-sso` does.

**To Reproduce**
1. Run `go-aws-sso` and complete flow
2. Run `go-aws-sso refresh`

**Expected behavior**
After completing auth flow via web during `refresh`, auth is refreshed.

**Error Output**
```
❯ go-aws-sso refresh --debug
2024-07-02 12:46:01 DEBUG go-aws-sso/main.go:305 Debug logging enabled
2024-07-02 12:46:01 DEBUG go-aws-sso/main.go:238 Checking mandatory flags
2024-07-02 12:46:01 DEBUG sso/aws.go:275 No lock file found
2024-07-02 12:46:01 DEBUG sso/aws.go:115 Encountered error while reading client information: no ClientInformation exist
2024-07-02 12:46:02 WARN sso/aws.go:172 Please verify your client request: https://device.sso.us-east-1.amazonaws.com/?user_code=REDACTED
2024-07-02 12:46:02 INFO sso/aws.go:234 Still waiting for authorization...
2024-07-02 12:46:05 INFO sso/aws.go:234 Still waiting for authorization...
2024-07-02 12:46:08 INFO sso/aws.go:234 Still waiting for authorization...
2024-07-02 12:46:12 INFO sso/aws.go:234 Still waiting for authorization...
2024-07-02 12:46:15 INFO internal/refresh.go:31 Using Start URL https://REDACTED.awsapps.com/start/#
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x99e7c9]

goroutine 1 [running]:
github.com/theurichde/go-aws-sso/internal.RefreshCredentials({0xf58778, 0xc00007c3e0}, {0xf59438, 0xc00007c400}, 0xc0000ab240)
	/home/paarth/go/pkg/mod/github.com/theurichde/go-aws-sso@v1.4.4-0.20240701021535-bf3c7bed6075/internal/refresh.go:37 +0x369
main.main.func2(0xc0000ab240)
	/home/paarth/go/pkg/mod/github.com/theurichde/go-aws-sso@v1.4.4-0.20240701021535-bf3c7bed6075/cmd/go-aws-sso/main.go:103 +0x54
github.com/urfave/cli/v2.(*Command).Run(0xc000444420, 0xc0000ab240, {0xc0004590a0, 0x2, 0x2})
	/home/paarth/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/command.go:276 +0x97d
github.com/urfave/cli/v2.(*Command).Run(0xc000444840, 0xc0000aaec0, {0xc000024330, 0x3, 0x3})
	/home/paarth/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/command.go:269 +0xbb7
github.com/urfave/cli/v2.(*App).RunContext(0xc000446000, {0xf55508, 0x13e7e80}, {0xc000024330, 0x3, 0x3})
	/home/paarth/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/app.go:333 +0x58b
github.com/urfave/cli/v2.(*App).Run(...)
	/home/paarth/go/pkg/mod/github.com/urfave/cli/v2@v2.27.2/app.go:307
main.main()
	/home/paarth/go/pkg/mod/github.com/theurichde/go-aws-sso@v1.4.4-0.20240701021535-bf3c7bed6075/cmd/go-aws-sso/main.go:172 +0xe26
```

**Additional context**

**Comment by theurichde** [comment]
Hi there! Thanks for filing the issue. 

I can not reproduce the issue on my side. 
Doing a complete flow and immediately doing a refresh works fine for me 🤔 

Do you wait some hours between the refresh?

**Comment by theurichde** [comment]
When I delete `.aws/sso/cache/last-usage.json` between both runs, I can provoke the error. 

I will have a look 🕵🏻‍♂️

**Comment by theurichde** [comment]
A fix is on its way. Classical nil pointer dereference.

I am wondering @PaarthShah, how it comes, that you get this error anyway? After a successful retrieval, it should write the usage information to `$HOME_DIR/.aws/sso/cache/last-usage.json` - can you confirm, that this file is written after a first successful credential flow?

**Comment by PaarthShah** [comment]
@theurichde I actually debugged it just a little bit ago; turns out the root cause was that `~/.aws/sso/cache` was owned by `root` instead of my user.

Which itself was a problem of my own making; I'd tried a docker-based solution before trying out `go-aws-sso` and I probably screwed up the permissions there 😅

A `chown -R` of that dir immediately fixed the issue for me.

I found that same part of the code, very nearly almost put up a PR but didn't clean it up yet: something about the log statement occurring before the actual error handling.
